### ISSUE TITLE
[BUG/ISSUE] HEMCO standalone 4x5 running error 

### ISSUE THREAD

**Issue by Jun-Meng** [geoschem/HEMCO#5]
# Report a HEMCO bug or technical issue

## Describe the bug
I’m using geosfp/merra2 4x5 HEMCO standalone to run online dust simulation. However, I got a HEMCO ERROR, which said that it can not assign mass tuning factor. I checked the source code in HEMCO/Extensions/hcox_dustdead_mod.F90 and found that the mass flux tuning factor “FLX_MSS_FDG_FCT” for GEOSFP 4X5 has already be set between lines 5526 and 5544. But the error indicated that the mass flux tuning factor “FLX_MSS_FDG_FCT” goes to -999 in lines 5670-5676.

## To Reproduce
1. Create a HEMCO standalone run directory using UT. 
2. Turn off all the emissions except dust (extension 105) in HEMCO_Config.rc

**Compilation commands**
1. make -j4 MET=merra2 GRID=4x5 CHEM=tropchem
2. copy CodeDir/bin/hemco_standalone.x  RunDir

**Run commands**
./hemco_standalone.x  HEMCO_sa_Config.rc

## Expected behavior
Run will stop and HEMCO error occur as indicated in the HEMCO.log

## Error messages

```
Using HEMCO v2.1.012
-------------------------------------------------------------------------------
 HEMCO grid definitions:
  --> Number of longitude cells:           72
  --> Number of latitude cells :           46
  --> Number of levels         :           47
  --> Lon range [deg E]        :   -182.500000000000        177.500000000000
  --> Lat range [deg N]        :   -90.0000000000000        90.0000000000000
Species DST4
Species DST3
Species DST2
Species DST1
-------------------------------------------------------------------------------
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Cannot assign mass flux tuning factor - please explicitly set it by adding ` --> Mass tuning factor: XX.X` to the extension options in your config. file.
ERROR LOCATION: HCOX_DustDead_GetFluxTun

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Error in getting dust DEAD mass tuning factor
ERROR LOCATION: HCOX_DustDead_Init (hcox_dustdead_mod.F90)
ERROR LOCATION: HCOX_INIT (hcox_driver_mod.F90)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Error encountered in routine "HCOX_Init"!
ERROR LOCATION: HCOI_SA_Init (in module HEMCO/Interfaces/hcoi_standalone_mod.F90)

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
HEMCO ERROR: Error encountered in routine "HCO_Sa_Init"!
ERROR LOCATION: HCOI_StandAlone_Run (in module HEMCO/Interfaces/hcoi_standalone_mod.F90

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

## Required information
Please include the following:
 - GEOS-Chem version you are using [e.g. 12.5.0]
 - Compiler version that you are using [e.g. ifort 16.0.4] 
 - netCDF and netCDF-Fortran library version
 - Computational environment [e.g. a cluster, Compute Canada Graham]
 - The Amazon Machine Image (AMI) ID that you used (if you ran on the AWS cloud)
 - Are you using "out of the box" code, or have you made modifications? 
          I used the default code. 

## Input and log files to attach

[HEMCO.log](https://github.com/geoschem/HEMCO/files/3817051/HEMCO.log)

## Additional context
I tested the 2x2.5 HEMCO standalone online dust run for both geosfp and merra2. Both of them work successfully.

**Comment by yantosca** [comment]
I think I have found the issue.  In the GEOS-Chem initialization routine HCOI_GC_INIT (in hcoi_gc_main_mod.F90), the values:

```F90
    ! Met and grid parameters
    iHcoConfig%MetField = Input_Opt%MetField
    iHcoConfig%GridRes  = State_Grid%GridRes
```

are set directly from the met field settings in input.geos.  But these seem to be undefined when you run the standalone.  I think these updates came in with the FlexGrid modifications.

In HEMCO_sa_Config.rc, we do have met field settings that you need to uncomment:
```
###############################################################
# Uncomment these lines if you are using GEOS-FP met fields!
METDIR:              /n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/GEOS_4x5/GEOS_FP
MET:                 GEOSFP
met:                 geosfp
CNYR:                2011
RES:                 4x5
NC:                  nc
###############################################################
# Uncomment these lines if you are using MERRA-2 met fields
#METDIR:              /n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/GEOS_4x5/MERRA2
#MET:                 MERRA2
#met:                 merra2
#CNYR:                2015
#RES:                 4x5
#NC:                  nc4
###############################################################
```
So the fix will involve passing the value of $MET to HcoState%Config%Metfield.  I'll start looking into this.

**Comment by Jun-Meng** [comment]
I believe that I uncommented those lines.

**Comment by yantosca** [comment]
Even if those lines are uncommented, they weren't used to define HcoState%Config%MetField and HcoState%Config%GridRes.   I was able to get the same error as you did.  Working on a fix now.

**Comment by yantosca** [comment]
This issue should now be resolved with commit https://github.com/geoschem/HEMCO/commit/f00c3093e9659f7a28cf61a8698d9027f9b9983f.  I was able to run the HEMCO standalone code after making this fix.  

We will also make the corresponding fix for GEOS-Chem 12.6.2.
